src/axom/sina/core/DataHolder.hpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SINA_DATAHOLDER_HPP
+#define SINA_DATAHOLDER_HPP
+
+/*!
+ ******************************************************************************
+ *
+ * \file DataHolder.hpp
+ *
+ * \brief   Implementation file for Sina DataHolder class
+ *
+ ******************************************************************************
+ */
+
+#include <string>
+#include <memory>
+#include <unordered_map>
+
+#include "conduit.hpp"
+
+#include "axom/sina/core/Datum.hpp"
+#include "axom/sina/core/CurveSet.hpp"
+
+namespace axom
+{
+namespace sina
+{
+
+/**
+ * \brief A DataHolder is a basic container for certain types of information.
+ *
+ * DataHolders contain curves, libraries, and data (Datum), and represent
+ * all the information a library can have associated with it. Records expand
+ * on DataHolders to contain additional info.
+ *
+ * \sa Record
+ * \sa LibraryDataMap
+ */
+class DataHolder
+{
+public:
+  /**
+     * An unordered map of Datum objects.
+     */
+  using DatumMap = std::unordered_map<std::string, Datum>;
```

Comment:
Similarly to my comment in `CurveSet` -- there might be benefits from leverage the collection hierarchies from sidre for `DatumMap`  (in the future).

Reply:
Great idea, from what I remember on the code integration side, there's quite a few concepts that map 1:1